### PR TITLE
Add TextFX

### PR DESCRIPTION
**Pull request by aaron-wade**
Add TextFX code

### CODE REVIEW

File: demos/palm/web/textfx/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+It contains 10 AI-powered 10 tools, each designed to explore creative possibilities with text and language.
+
+![TextFX demo gif](https://storage.googleapis.com/textfx-assets/demo.gif)
+
+A live hosted version of _TextFX_ is available at [textfx.withgoogle.com](https://textfx.withgoogle.com/).
+
+This experiment is an example of how you can use the PaLM API to build applications that leverage Google's state of the art large language models (LLMs).
+
+The PaLM API consists of two services, each with a distinct method for generating content:
```
Comment (markmcd) [inline demos/palm/web/textfx/README.md:27]: > The PaLM API consists of two services

This is false now, and will become increasingly more false as we add more services. No need to list things you don't use - just drop references to Chat if they're irrelevant.

File: demos/palm/web/textfx/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+_TextFX_ uses this same strategy to prime the model to perform the tasks associated with each of its tools. You can find all the prompts in `src/lib/priming.js`, or view them in the app by clicking "Look Under the Hood."
+
+## Requirements
+
+- Node.js (version 18.15.0 or higher)
+- Firebase project
+
+Make sure you have `npm` or `yarn` set up on your machine.
+
+## Developer Setup
```
Comment (markmcd) [inline demos/palm/web/textfx/README.md:63]: Thanks for adding detailed setup steps

**Review by markmcd — APPROVED** [review]
Project looks fun :)

Let me know what test automation you need set up, or if you'd prefer to handle dependabot requests manually.

File: demos/palm/web/textfx/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+7. Return to https://console.cloud.google.com and select your Firebase project. Click _More Products_ at the bottom of the left-side menu, then scroll down and click _Cloud Functions_. Select each `ext-palm-secure-backend` Cloud Function and then click _Permissions_ at the top. Add `allUsers` to the Cloud Functions Invoker role.
+
+The above instructions assume that your Firebase backend will be used for individual/experimental purposes. If you anticipate broader usage, consider enabling App Check (see https://firebase.google.com/docs/app-check for an in-depth implementation guide).
+
+To run the application locally:
+
+1. Clone this repo to your local machine.
+
+2. Run `npm i` or `yarn` in the root folder to install dependencies.
```
Comment (markmcd) [inline demos/palm/web/textfx/README.md:91]: This didn't work for me straight up.  Husky didn't like being installed when the .git dir isn't adjacent: `husky - .git can't be found`.

I got around it with `ln -s ../../../../.git` in the textfx dir.

File: demos/palm/web/textfx/README.md (hunk 1 above)
Comment (aaron-wade) [reply demos/palm/web/textfx/README.md:27]: Got it! Will update in a separate PR.

File: demos/palm/web/textfx/README.md (hunk 3 above)
Comment (aaron-wade) [reply demos/palm/web/textfx/README.md:91]: Gotcha—I think this dependency can actually be removed, as I didn't copy over any of the git hooks. Will update in a separate PR.

File: demos/palm/web/textfx/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+3. Add your Firebase info to `src/lib/firebase.config.js`.
```
[the remark this answers is not in the document]
Comment (aaron-wade) [reply demos/palm/web/textfx/README.md:93]: The entire config object containing all of the required fields for your Firebase project can be found in the Firebase console—the steps for finding it are outlined in firebase.config.js. As long as you replace the entire config object in firebase.config.js with the one associated with your project, you should be good to go :)

**Pull request merged**